bin/ch/WScriptJsrt.cpp

```diff
@@ -1371,6 +1372,48 @@ JsValueRef __stdcall WScriptJsrt::SleepCallback(JsValueRef callee, bool isConstr
     return returnValue;
 }
 
+JsValueRef __stdcall WScriptJsrt::GetProxyPropertiesCallback(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+    HRESULT hr = E_FAIL;
+    JsValueRef returnValue = JS_INVALID_REFERENCE;
+    JsErrorCode errorCode = JsNoError;
+
+    IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&returnValue));
+
+    if (argumentCount > 1)
+    {
+        bool isProxy = false;
+        JsValueRef target;
+        JsValueRef handler;
+        ChakraRTInterface::JsGetProxyProperties(arguments[1], &isProxy, &target, &handler);
+
+        if (isProxy)
+        {
+            JsPropertyIdRef targetProperty;
+            JsPropertyIdRef handlerProperty;
+            
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("target", &targetProperty));
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("handler", &handlerProperty));
+            IfJsrtErrorSetGo(ChakraRTInterface::JsCreateObject(&returnValue));
+
+            if (target == nullptr)
+            {
+                JsValueRef undefined = JS_INVALID_REFERENCE;
+                IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&undefined));
```

Comment:
We already called `JsGetUndefinedValue` above. Can we fold these two into one call at the top?